src/libs/NumberUtils.js

```diff
@@ -0,0 +1,26 @@
+import CONST from "../CONST";
+
+/**
+ * Generates a random positive 64 bit integer by randomly generating the left half and right half and concatenating them
+ * @returns {string} randomly generated 64 bit string
+ */
+function rand64() {
+    // Max 64-bit signed:
+    // 9,223,372,036,854,775,807
+    // The top part of the max 64-bit number *+1* because we're flooring it
+    let top = Math.floor(Math.random() * CONST.MAX_64BIT_TOP_HALF + 1);
+  
+    // The bottom part of the max 64-bit number *+1* for the same reason
+    let rest = Math.floor(Math.random() * CONST.MAX_64BIT_BOTTOM_HALF + 1);
+  
+    // If the top is any number but the highest one, we can actually have any value for the rest
+    if (top != CONST.​​MAX_64BIT_TOP_HALF) {
+      rest = Math.floor(Math.random() * 1000000000);
```

Comment:
@flodnv @luacmartins you can double check my snake challenge knowledge, but if we want to skip the first 1000000 for reportIDs (or more for reportActionIDs) we could:

1. Change the method to rand64(minimum)
2. Update this to `Math.floor(Math.random() * (1000000000 - minimum)) + minimum`
3. Update the other bottomHalf to `Math.floor(Math.random() * (CONST.MAX_64BIT_BOTTOM_HALF - minimum) + 1) + minimum`;

Reply:
And then when getting an ID for reportIDs just pass `100000000`, transactions will pass its minimum based on our current max in the DB, etc

Reply:
Would this way work? Let's say the left half is 8223372036, if minimum is 1000000, then you wouldn't be able to reach 8223372036 099999999. 

Also, if min is 1000000000, then rest would always be 0 which should not be the case I think.

Reply:
I actually think we don't need to worry about using a minimum.  We are not doing it for transactionIDs and we have a lot more transactions in the DB.

Since the 64bit distribution is so sparse, the 85m reports we have make almost no difference in increasing the chances of a collision.

Reply:
> ? Let's say the left half is 8223372036, if minimum is 1000000, then you wouldn't be able to reach 8223372036 099999999 for example because rest is always greater than minimum which is 1000000 in this case.

Thinking through this:
1. We have left = 8223372036
2. We generate a number between 0 and 999,000,000 (so max would be 998,999,999)
3. We add 1,000,000 making the max right side 999,999,999, and min right side 1,000,000

> Also, if min is 1000000000, then rest would always be 0 which should not be the case I think.

Good point about the minimum being higher than the right side (it is for reportActionID), we will need conditional logic for that.

Reply:
@luacmartins we have over a billion transactionIDs. I know that's still small compared to 9 quintillion, but it's not nothing /shrug

Reply:
I'm trying to come up with a good way but I can't yet, here's my thoughts. The idea is to split the string into the left half, the left of the minimum and then the rest of the right half. Then we could use Brandon's logic of subtracting the min, multiplying it by the random number and adding the min back to the right half. But the problem with this is if the minimum is so big it goes into the right half, then we have the same initial problem since we can only generate 32 bit random numbers. To fix this we could add a max minimum which is equal to CONST.MAX_64BIT_BOTTOM_HALF (right half). I thought this was hacky but I think we have to have this condition regardless even if we do it the other way. The other way I believe I read in the thread is to generate random numbers until one is greater than the min. But to do this, the min has to be below the bottom half since if it goes into the top half, then we are comparing a >32 bit number with a >32 bit number which I don't think is natively possible. I'm re-reading this and I realized it's kind of confusing so let me know if I can elaborate on any portion.

Basically, my proposal is to add a constraint that the min has to be less than the CONST.MAX_64BIT_BOTTOM_HALF and then split the right string into two parts and if the left part is all 0's then do (rand * (right half magnitude - min)) + min and concatenate all three portions.

Reply:
Can't we just do:

```
if (minimum > MAX_64BIT_BOTTOM_HALF) {
   leftMinimum = minimum.toString().substring(0,minimum.toString().length - 9);
   // code I had up there but using leftMinimum instead of minimum
} else {
   // code I had up there
}
```

Reply:
Well 1 billion is only 0.0000000108% of the whole available range. I think the chances of having a collision with or without setting a minimum would not change much 🤷 

We decided against it in this thread - https://docs.google.com/document/d/125xZPDi7v6jmS-GKEcvcpdAwpDRPPGsq_iEOCl9RQFw/edit?disco=AAAAdh4GcbM

Reply:
Okay let's just not bother then. Sorry for the confusion @srikarparsi